File: mario.c
#include<stdio.h>
#include<cs50.h>

int main(void)
{
    int height, row, column, space;
    do 
    {
        height = get_int("enter heigt here: ");
    }
    while (height < 1 || height > 8);  //the users height input
    
    
    for (row = 0; row < height; row++)// row loop is the loop that all they anthor loops is nested below it  and it adds new lines.
    {
        
        
        for (column = 0; column <= row ; column++) // adds # 
        {
            printf("#");
        }
    
        printf("\n");
    }
    
    

    
}

File: mario.py
from cs50 import get_int
# user`s input
while True:
    n = get_int("Height: ")
    if n >= 1 and n <= 8:
        break
k = 1  # icrement of hashes
s = n - 1  # num of spaces
# all of the nested loops

for i in range(1, n+1, 1):
    for s in range(s):
        print(' ', end='')
    s -= 1   
    for j in range(k):
        print("#", end='')
    k += 1
    s += 1
    print()